tensorflow_datasets/image/imagewang.py

```diff
@@ -0,0 +1,142 @@
+
+# TODO(imagewang): BibTeX citation
+# See https://github.com/fastai/imagenette/issues/16.
+_CITATION = """
+"""
+
+_DESCRIPTION = """\
+Imagewang contains Imagenette and Imagewoof combined
+Image网 (pronounced "Imagewang"; 网 means "net" in Chinese) contains Imagenette
+and Imagewoof combined, but with some twists that make it into a tricky
+semi-supervised unbalanced classification problem:
+
+* The validation set is the same as Imagewoof (i.e. 30% of Imagewoof images);
+  there are no Imagenette images in the validation set (they're all in the
+  training set)
+* Only 10% of Imagewoof images are in the training set!
+* The remaining are in the unsup ("unsupervised") directory, and you can not
+  use their labels in training!
+* It's even hard to type and hard to say!
+
+The dataset comes in three variants:
+  * Full size
+  * 320 px
+  * 160 px
+This dataset consists of the Imagenette dataset {size} variant.
+"""
+
+_LABELS_FNAME = "image/imagewang_labels.txt"
+_URL_PREFIX = "https://s3.amazonaws.com/fast-ai-imageclas"
+_SIZES = ["full-size", "320px", "160px"]
+
+_SIZE_TO_DIRNAME = {
+    "full-size": "imagewang",
+    "320px": "imagewang-320",
+    "160px": "imagewang-160"
+}
+
+
+class ImagewangConfig(tfds.core.BuilderConfig):
+  """BuilderConfig for Imagewang."""
+
+  def __init__(self, size, **kwargs):
+    super(ImagewangConfig, self).__init__(
+        version=tfds.core.Version("0.1.0"), **kwargs)
+    self.size = size
+
+
+def _make_builder_configs():
+  configs = []
+  for size in _SIZES:
+    configs.append(
+        ImagewangConfig(
+            name=size,
+            size=size,
+            description=_DESCRIPTION.format(size=size)))
+  return configs
+
+
+class Imagewang(tfds.core.GeneratorBasedBuilder):
+  """ Imagewang contains Imagenette and Imagewoof combined. """
+  
+  VERSION = tfds.core.Version("0.1.0")
+  BUILDER_CONFIGS = _make_builder_configs()
+
+  def _info(self):
+    names_file = tfds.core.get_tfds_path(_LABELS_FNAME)
+    return tfds.core.DatasetInfo(
+        builder=self,
+        description=_DESCRIPTION,
+        features=tfds.features.FeaturesDict({
+            "image": tfds.features.Image(),
+            "label": tfds.features.ClassLabel(names_file=names_file)
+        }),
+        supervised_keys=("image", "label"),
+        homepage='https://dataset-homepage://github.com/fastai/imagenette',
+        citation=_CITATION,
+    )
+
+  def _split_generators(self, dl_manager):
+    """Returns SplitGenerators."""
+    size = self.builder_config.size
+    if size in _SIZES:
+      size_str = "" if size == "full-size" else "-" + size[:-2]
+      url = os.path.join(_URL_PREFIX, "imagewang%s.tgz" % size_str)
+      print(url)
+      path = dl_manager.download_and_extract(url)
+      train_path = os.path.join(path, _SIZE_TO_DIRNAME[size], "train")
+      val_path = os.path.join(path, _SIZE_TO_DIRNAME[size], "val")
+    else:
+      raise ValueError("Size not implemented!")
```

Comment:
Can this be changed to something more informative, i.e, `raise ValueError("size must be one of %s" % _SIZES)`?